src/components/Modal/ModalHeader.tsx

```diff
@@ -0,0 +1,59 @@
+import React, { forwardRef } from 'react'
+
+import cn from 'classnames'
+
+import { IconRemove } from 'components/Icons'
+
+import styles from './ModalHeader.module.css'
+import { ModalHeaderProps, ModalTitleProps } from './types'
+
+/**
+ * Header component to be used inside modals
+ */
+export const ModalHeader = forwardRef<HTMLDivElement, ModalHeaderProps>(
+  function ModalHeader(
+    { className, onClose, showDismissButton = true, children, ...props },
+    ref
+  ) {
+    return (
+      <div
+        className={cn(styles.headerContainer, className)}
+        ref={ref}
+        {...props}
+      >
+        {showDismissButton && (
+          <div className={styles.dismissButton} onClick={onClose}>
+            <IconRemove />
+          </div>
+        )}
+        {children}
+      </div>
+    )
+  }
+)
+
+/**
+ * Title component to be used inside modal headers
+ */
+export const ModalTitle = forwardRef<HTMLDivElement, ModalTitleProps>(
+  function ModalHeader(
+    { subtitleClassName, icon, iconClassName, title, subtitle, ...props },
+    ref
+  ) {
+    return (
+      <>
+        <div className={styles.title} {...props} ref={ref}>
+          {icon == null ? null : (
```

Comment:
i think no need for null check, unless icon being undefined is a case we want to catch?

Reply:
okay so if its not provided, then we still want to render the icon containing div? cause optional makes it `undefined` and it would only be null if someone was like `<ModalHeader icon={null} ... />` i believe